internal/build/imgsrc/ensure_builder.go

```diff
@@ -19,7 +18,67 @@ import (
 	"go.opentelemetry.io/otel/trace"
 )
 
-func EnsureBuilder(ctx context.Context, org *fly.Organization, region string, recreateBuilder bool) (*fly.Machine, *fly.App, error) {
+type Provisioner struct {
+	org           *fly.Organization
+	useVolume     bool
+	buildkitAddr  string
+	buildkitImage string
+}
+
+func NewProvisioner(org *fly.Organization) *Provisioner {
+	return &Provisioner{
+		org:       org,
+		useVolume: true,
+	}
+}
+
+func NewBuildkitProvisioner(org *fly.Organization, addr, image string) *Provisioner {
+	return &Provisioner{
+		org:           org,
+		useVolume:     true,
+		buildkitAddr:  addr,
+		buildkitImage: image,
+	}
+}
+
+func (p *Provisioner) UseBuildkit() bool {
+	return p.buildkitAddr != "" || p.buildkitImage != ""
+}
+
+const defaultImage = "docker-hub-mirror.fly.io/flyio/rchab:sha-9346699"
+
+func (p *Provisioner) image() string {
```

Comment:
I pushed a buildkit image with embedded mirror config we can use to `flyio/buildkit`, so we could use a hard-coded alias for `default` or something similar:

```suggestion
const defaultBuildkitImage = "docker-hub-mirror.fly.io/flyio/buildkit@sha256:0fe49e6f506f0961cb2fc45d56171df0e852229facf352f834090345658b7e1c"

func (p *Provisioner) image() string {
	if p.buildkitImage == "default" {
		return defaultBuildkitImage
	}
```

Reply:
Are you going to remove -buildkit-addr evntually? Right now, the flag and -buildkit-image are used for enabling BuildkitBuilder

Reply:
> Are you going to remove -buildkit-addr evntually? Right now, the flag and -buildkit-image are used for enabling BuildkitBuilder

no, being able to specify an endpoint directly is helpful for use-cases where the user may want to manually manage their own buildkit service.